services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedRegistrationService.java

```diff
@@ -508,4 +521,12 @@ private static JsonArray getSearchDevicesAggregatePipelineQuery(final String ten
                     return Future.succeededFuture();
                 });
     }
+
+    private Future<Void> tenantExists(final String tenantId, final Span span) {
+
+        return tenantInformationService.tenantExists(tenantId, span)
+                .compose(result -> result.isError()
+                        ? Future.failedFuture(new ClientErrorException(result.getStatus()))
```

Comment:
IMHO you should not assume the status code to be in the [400, 500) range here. Instead, use *StatusCodeMapper.from(int, String)* which takes care of this ...
For readability I would also explicitly check for OK instead, i.e.
```java
result.isOK() ? Future.succeededFuture() : StatusCodeMapper.from(result.getStatus(), null);
```

Reply:
Thanks for pointing it out. I have changed it now and pushed a commit addressing all the comments.